ui/app/components/app/connected-accounts-list/connected-accounts-list.component.js

```diff
@@ -0,0 +1,137 @@
+import { DateTime } from 'luxon'
+import PropTypes from 'prop-types'
+import React, { PureComponent } from 'react'
+import ConnectedAccountsListPermissions from './connected-accounts-list-permissions'
+import ConnectedAccountsListItem from './connected-accounts-list-item'
+import ConnectedAccountsListOptions from './connected-accounts-list-options'
+import ConnectedAccountsListOptionsItem from './connected-accounts-list-options-item'
+
+export default class ConnectedAccountsList extends PureComponent {
+  static contextTypes = {
+    t: PropTypes.func.isRequired,
+  }
+
+  static defaultProps = {
+    accountToConnect: null,
+    permissions: undefined,
+  }
+
+  static propTypes = {
+    accountToConnect: PropTypes.shape({
+      address: PropTypes.string.isRequired,
+      name: PropTypes.string.isRequired,
+    }),
+    connectedAccounts: PropTypes.arrayOf(PropTypes.shape({
+      address: PropTypes.string.isRequired,
+      name: PropTypes.string.isRequired,
+      lastActive: PropTypes.number.isRequired,
+    })).isRequired,
+    permissions: PropTypes.arrayOf(PropTypes.shape({
+      key: PropTypes.string.isRequired,
+      description: PropTypes.string.isRequired,
+    })),
+    addPermittedAccount: PropTypes.func.isRequired,
+    removePermittedAccount: PropTypes.func.isRequired,
+    setSelectedAddress: PropTypes.func.isRequired,
+  }
+
+  state = {
+    optionsAddress: null,
+  }
+
+  toggleOptions = (address) => () => {
+    this.setState((prevState) => ({
+      optionsAddress: prevState.optionsAddress === address
+        ? null
+        : address,
+    }))
+  }
+
+  connectAccount = () => {
+    const { optionsAddress: address } = this.state
+
+    this.props.addPermittedAccount(address)
+  }
+
+  disconnectAccount = () => {
+    const { optionsAddress: address } = this.state
+
+    this.props.removePermittedAccount(address)
+  }
+
+  switchAccount = () => {
+    const { optionsAddress: address } = this.state
+
+    this.props.setSelectedAddress(address)
+  }
+
+  renderUnconnectedAccount () {
+    const { accountToConnect } = this.props
+    const { t } = this.context
+    const { optionsAddress } = this.state
+
+    if (!accountToConnect) {
+      return null
+    }
+
+    const { address, name } = accountToConnect
+    return (
+      <ConnectedAccountsListItem
+        className="connected-accounts-list__row--highlight"
+        address={address}
+        name={`${name} (…${address.substr(-4, 4)})`}
+        status={(
+          <>
+            {t('statusNotConnected')}
+            &nbsp;&middot;&nbsp;
+            <a className="connected-accounts-list__account-status-link">{t('connect')}</a>
```

Comment:
Presumably this should connect the account when clicked. It looks like it doesn't do that right now.